near-sdk-sim/src/user.rs

```diff
@@ -154,18 +154,35 @@ impl UserAccount {
 
     /// Make a contract call.  `pending_tx` includes the receiver, the method to call as well as its arguments.
     /// Note: You will most likely not be using this method directly but rather the [`call!`](./macro.call.html) macro.
-    pub fn call(
+    pub fn function_call(
         &self,
         pending_tx: PendingContractTx,
         deposit: Balance,
         gas: Gas,
     ) -> ExecutionResult {
-        self.submit_transaction(self.transaction(pending_tx.receiver_id).function_call(
-            pending_tx.method.to_string(),
-            pending_tx.args,
-            gas,
-            deposit,
-        ))
+      self.call(pending_tx.receiver_id.clone(),
+          &pending_tx.method,
+          &pending_tx.args,
+          gas,
+          deposit,
+      )
+        
+    }
+
+    pub fn call(
+      &self,
+      receiver_id: AccountId,
+      method: &str, 
+      args: &[u8],
+      gas: Gas,
+      deposit: Balance,
```

Comment:
Could we make these `Option`s so I can pass `None, None` to get the defaults?

Reply:
Well, `None` should not mean "default". If you want something default in Rust, you have it - `Default` trait (`Gas::default()`, `Balance::default()`), though I would still consider explicit useful constants instead of `default` since I cannot really tell what is the right "default balance" or "default gas", so I suggest we introduce some baseline consts in near-sdk. The default deposit to a function call is usually zero.

Reply:
We already have two  `near_sdk_sim::DEFAULT_GAS` and `near_sdk_sim::STORAGE_AMOUNT`. Personally I think it's best to just use defaults in the `call/deploy` macros.